src/main/java/de/medizininformatik_initiative/processes/common/fhir/client/AsyncFhirClientImpl.java

```diff
@@ -0,0 +1,95 @@
+{
+	private static final Logger logger = LoggerFactory.getLogger(AsyncFhirClientImpl.class);
+
+	private final int initialPollingIntervalMilliseconds;
+
+	public AsyncFhirClientImpl(KeyStore trustStore, KeyStore keyStore, char[] keyStorePassword, int connectTimeout,
+			int socketTimeout, String fhirServerBasicAuthUsername, String fhirServerBasicAuthPassword,
+			String fhirServerBearerToken, TokenProvider fhirServerOAuth2TokenProvider, String fhirServerBase,
+			String proxyUrl, String proxyUsername, String proxyPassword, int initialPollingIntervalMilliseconds,
+			FhirContext fhirContext, String localIdentifierValue, DataLogger dataLogger)
+	{
+		super(trustStore, keyStore, keyStorePassword, connectTimeout, socketTimeout, fhirServerBasicAuthUsername,
+				fhirServerBasicAuthPassword, fhirServerBearerToken, fhirServerOAuth2TokenProvider, fhirServerBase,
+				proxyUrl, proxyUsername, proxyPassword, fhirContext, localIdentifierValue, dataLogger);
+
+		this.initialPollingIntervalMilliseconds = initialPollingIntervalMilliseconds;
+	}
+
+	@Override
+	public Resource search(String url)
+	{
+		HttpClient client = createClient();
+		HttpRequest request = createBaseRequest(url).header("Prefer", "respond-async").GET().build();
+
+		try
+		{
+			logger.debug("Async search for URL '{}' started", url);
+			HttpResponse<InputStream> response = client.send(request, HttpResponse.BodyHandlers.ofInputStream());
+
+			int currentPollingIntervalMilliseconds = initialPollingIntervalMilliseconds;
+			while (response.statusCode() == HttpURLConnection.HTTP_ACCEPTED)
+			{
+				response.body().close();
+				response = pollSearchResultAfterDelay(client, currentPollingIntervalMilliseconds, response.headers(),
+						url);
+				currentPollingIntervalMilliseconds = currentPollingIntervalMilliseconds * 2;
+			}
+
+			if (response.statusCode() == HttpURLConnection.HTTP_OK)
+			{
+				try (InputStream body = response.body())
+				{
+					return (Resource) getFhirContext().newJsonParser().parseResource(body);
+				}
+			}
+			else
+			{
+				response.body().close();
+				throw new RuntimeException(
+						"Request for URL '" + url + "' failed - status code: " + response.statusCode());
+			}
+		}
+		catch (Exception exception)
+		{
+			throw new RuntimeException("Async search for URL '" + url + "' failed - " + exception.getMessage(),
+					exception);
+		}
+	}
+
+	private HttpResponse<InputStream> pollSearchResultAfterDelay(HttpClient client, int pollingInterval,
+			HttpHeaders headers, String url) throws IOException, InterruptedException
+	{
+		logger.debug("Async search for '{}' in-progress, checking result in {} milliseconds", url,
+				initialPollingIntervalMilliseconds);
+		Thread.sleep(pollingInterval);
+
+		String location = headers.firstValue("Content-Location")
+				.orElseThrow(() -> new RuntimeException("No Content-Location header returned"));
+		String locationPath = location.substring(location.indexOf("__async-status"));
```

Comment:
The path prefix `__async-status` is blaze specific as the [documentation](http://hl7.org/fhir/R5/async-bundle.html#3.2.6.2.2.0.2) does not state a specific URI form.

Reply:
Is it not equal to just keeping everything after the first `/`?

Reply:
Using now the provided FHIR server baseURL to check wether the Content-Location starts with the same URL: 0ad23ab